tendrl/commons/flows/create_cluster/__init__.py

```diff
@@ -232,3 +232,10 @@ def run(self):
         finally:
             # releasing nodes if any exception came
             create_cluster_utils.release_node_lock(self.parameters)
+
+            # Persist the cluster's extra details
+            NS.tendrl.objects.Cluster(
+                integration_id=self.parameters['TendrlContext.integration_id'],
+                public_network=self.parameters.get('Cluster.public_network', ""),
+                cluster_network=self.parameters.get('Cluster.cluster_network', "")
```

Comment:
In case of gluster cluster creation only cluster_network value is available in parameters. May be we can add mandatory availability of cluster_network at least. Also by this time as cluster creation is done so at least cluster_network should be available.